client/me/account/main.jsx

```diff
@@ -84,21 +82,50 @@ const Account = React.createClass( {
 		debug( this.constructor.displayName + ' component is unmounting.' );
 	},
 
-	updateLanguage() {
-		let valueLink = this.valueLink( 'language' );
+	getUserSetting( settingName ) {
+		return this.props.userSettings.getSetting( settingName );
+	},
 
-		valueLink.requestChange = ( value ) => {
-			const originalLanguage = this.props.userSettings.getOriginalSetting( 'language' );
+	updateUserSetting( settingName ) {
+		return event => this.props.userSettings.updateSetting( settingName, event.target.value );
+	},
 
-			this.props.userSettings.updateSetting( 'language', value );
-			if ( value !== originalLanguage ) {
-				this.setState( { redirect: '/me/account' } );
-			} else {
-				this.setState( { redirect: false } );
-			}
-		};
+	updateUserSettingCheckbox( settingName ) {
+		return event => this.props.userSettings.updateSetting( settingName, event.target.checked );
+	},
+
+	updateLanguage( event ) {
+		const { value } = event.target;
+		const originalLanguage = this.props.userSettings.getOriginalSetting( 'language' );
 
-		return valueLink;
+		this.props.userSettings.updateSetting( 'language', value );
+		if ( value !== originalLanguage ) {
+			this.setState( { redirect: '/me/account' } );
+		} else {
+			this.setState( { redirect: false } );
+		}
+	},
+
+	getEmailAddress() {
+		return this.hasPendingEmailChange()
+			? this.props.userSettings.getSetting( 'new_user_email' )
+			: this.props.userSettings.getSetting( 'user_email' );
+	},
+
+	updateEmailAddress( event ) {
+		const { value } = event.target;
+		if ( '' === value ) {
```

Comment:
In this case, the `event.target` is a `HTMLInputElement` and its `value` property is guaranteed to be a `DOMString` - it's never `null` or anything else.